cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -3834,7 +3834,55 @@ macro(build_grpc)
 
   list(APPEND
        ARROW_BUNDLED_STATIC_LIBS
-       ${GRPC_GPR_ABSL_LIBRARIES}
+       # This list of absl dependencies can be generated with:
+       # pkg-config --libs --static grpc \
+       #   | tr " " "\n" \
+       #   | grep ^-labsl_ \
+       #   | sed 's/^-labsl_/absl::/'
```

Comment:
Maybe state where should this be run?

Reply:
It should be able to be run anywhere as long as gRPC is on your `PKG_CONFIG_PATH`. If you have installed grpc via package manager then it will be. Otherwise you might have to add a build directory to that path. If you try to run the command and `grpc` isn't on the path, `pkg-config` will provide an error message saying it's not found and you should check your `PKG_CONFIG_PATH`.

Reply:
Why did you replace `${GRPC_GPR_ABSL_LIBRARIES}` instead of updating `set(GRPC_GPR_ABSL_LIBRARIES ...)`?

Reply:
hmm I suppose I should have just done that. I was thinking that `GRPC_GPR_ABSL_LIBRARIES` was meant to hold the top-level Abseil dependencies of gRPC, while for the static library list we need a flattened set of libraries. That is, the static library list not only needs to contain `absl::cord`, but also `absl::cord_zinfo`, but most other functions using the target can find the nested dependency automatically. But I guess there's no reason not to use a flattened list everywhere.